pkg/pipelines/add_nodes.go

```diff
@@ -53,11 +54,11 @@ func NewAddNodesPipeline(runtime *common.KubeRuntime) error {
 		&kubernetes.StatusModule{},
 		&container.InstallContainerModule{},
 		&images.PullModule{Skip: runtime.Arg.SkipPullImages},
-		&etcd.PreCheckModule{},
-		&etcd.CertsModule{},
-		&etcd.InstallETCDBinaryModule{},
-		&etcd.ConfigureModule{},
-		&etcd.BackupModule{},
+		&etcd.PreCheckModule{Skip: runtime.Cluster.Etcd.Type != kubekeyapiv1alpha2.KubeKey},
+		&etcd.CertsModule{Skip: runtime.Cluster.Etcd.Type != kubekeyapiv1alpha2.KubeKey},
```

Comment:
If we add master nodes and use an `external` etcd type, this module will be skipped.

Reply:
OK, I got it. I will adjust it here.

Reply:
Done